### PR TITLE
api: node cpu mem disk networkd stats

### PR DESCRIPTION
**Pull request by yangwenmai**
stat node static cpu, mem, disk, network info.
stat real time cpu, mem, netcard info.

### What is changed and how it works?

add `nodeutil` to get static and dynamic stats info.

### Check List 

Tests

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has HTTP API interfaces change

### CODE REVIEW

**Comment by disksing** [comment]
Hi @yangwenmai , thank for you contribution. We do not use `feat: ...` style commit message or PR title. Please follow the style described in doc: https://github.com/pingcap/pd/blob/master/CONTRIBUTING.md#format-of-the-commit-message

File: pkg/nodeutil/stats_info.go [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+	var nodeIPs []string
+	nodes := getPDNode(svr, existNodeHost)
+	if len(nodes) > 0 {
+		nodeIPs = append(nodeIPs, nodes...)
+	}
+	nodes = getTiKVNode(svr, existNodeHost)
+	if len(nodes) > 0 {
+		nodeIPs = append(nodeIPs, nodes...)
+	}
+	nodes = getTiDBNode(svr, existNodeHost)
```
Comment (siddontang) [inline pkg/nodeutil/stats_info.go]: I don't like introducing too many TiDB codes into PD. 

What do you think @disksing

Comment (disksing) [reply pkg/nodeutil/stats_info.go]: Agree. Why not send all server IPs from the client?

I think there is some confusion at the abstract level. This PR seems to want to query the status of a single node (including CPU, memory, disk, and network connectivity to other nodes). I assume that all nodes in the cluster should be able to provide a similar interface. The problem is other than the PD, other nodes have no similar method to get the information of all nodes locally. So it seems reasonable to provide a list of all nodes by the client.

**Comment by lonng** [comment]
@yangwenmai Hi,  I think we can retrieve latency from Prometheus and don't need to add the logic to PD.
We prepare to implement a framework in TiDB which will projection a Prometheus expression to a virtual table.

**Comment by siddontang** [comment]
> We prepare to implement a framework in TiDB which will projection a Prometheus expression to a virtual table.

If we want to do this internally, don't depend on Prometheus. Why not let TiDB do the Ping itself? The TiDB can know all other TiDBs, TiKVs and PDs.

**Comment by disksing** [comment]
@siddontang I think they want ping latency of every 2 nodes (NxN).

**Comment by lonng** [comment]
I think we can close this issue because this feature has been included in `sysutil` dependence.

**Comment by yangwenmai** [comment]
ok

**Comment by rleungx** [comment]
I am going to close it according to the above comment. If there is any problem after, feel free to reopen it.

**Pull request closed without merging**